scripts/distribute-into-proxies/index.js

```diff
@@ -0,0 +1,201 @@
+  ['168750000', vesting],
+  ['146250000', vesting],
+  ['112500000', vesting],
+  ['112500000', vesting],
+  ['112500000', vesting],
+  ['112500000', vesting],
+  ['112500000', vesting],
+  ['45000000', vesting],
+  ['22500000', vesting],
+  ['11625000000', vesting],
+]
+
+const total = allocation
+  .reduce((acc, [amount]) => acc.plus(amount), new BigNumber(0))
+  .multipliedBy(UNIT)
+  .toFixed()
+log('total to be distributed:', total)
+
+function calculateSchedule([amount, { start, period, period_count }]) {
+  const total = new BigNumber(amount).multipliedBy(UNIT)
+
+  const per_period = total
+    .div(period_count)
+    .decimalPlaces(0, BigNumber.ROUND_FLOOR)
+    .toFixed()
+  const remainder = total.mod(period_count).toFixed()
+
+  return {
+    remainder,
+    schedule: {
+      start,
+      period,
+      per_period,
+      period_count,
+    },
+  }
+}
+
+const distribution = allocation.map(calculateSchedule)
+
+const totalDistributed = distribution
+  .reduce(
+    (acc, { schedule: { per_period, period_count }, remainder }) =>
+      acc
+        .plus(remainder)
+        .plus(new BigNumber(per_period).multipliedBy(period_count)),
+    new BigNumber(0),
+  )
+  .toFixed()
+
+assert.equal(total, totalDistributed, 'total distributed does not match')
+
+const bsxAddress = (pubKey) => encodeAddress(pubKey, 10041)
+const sendAndWait = (from, tx, nonce = -1) =>
+  new Promise(async (resolve, reject) => {
+    try {
+      await tx.signAndSend(from, { nonce }, (receipt) => {
+        if (receipt.status.isInBlock) {
+          resolve(receipt)
+        }
+      })
+    } catch (e) {
+      reject(e)
+    }
+  })
+
+async function main() {
+  await cryptoWaitReady()
+  const provider = new WsProvider(RPC)
+  const keyring = new Keyring({ type: 'sr25519' })
+  const api = await ApiPromise.create({ provider })
+  const [chain, nodeVersion] = await Promise.all([
+    api.rpc.system.chain(),
+    api.rpc.system.version(),
+  ])
+  log(`connected to ${RPC} (${chain} ${nodeVersion})`)
+  const from = keyring.addFromUri(ACCOUNT_SECRET)
+  const activeAccount = bsxAddress(from.addressRaw)
+  log('active account:', activeAccount)
+  const treasuryPubKey = stringToU8a('modlpy/trsry'.padEnd(32, '\0'))
+  const treasury = bsxAddress(treasuryPubKey)
+  log('treasury account:', treasury)
+
+  log('creating anonymous proxies...')
+  const proxies = distribution.map(() =>
+    api.tx.proxy.anonymous('Any', 0, proxyIndex++),
+  )
+  const receipt1 = await sendAndWait(from, api.tx.utility.batchAll(proxies))
+  const anonymousProxies = receipt1.events
+    .filter(({ event }) => event.method === 'AnonymousCreated')
+    .map(({ event }) => event.data.anonymous.toHuman())
+  assert.equal(
+    anonymousProxies.length,
+    distribution.length,
+    'not all proxies created',
+  )
+  log('proxies created:', anonymousProxies)
+
+  log('funding proxies...')
+  const transfers = anonymousProxies.map((anon) =>
+    api.tx.balances.transfer(anon, 500 * UNIT),
```

Comment:
Is this enough after fee bump? We probably also need sudo here if we do it b4 upgrade.

Reply:
its for storage deposit, which shouldn't change, but we'll test